code/modules/surgery/organs/heart.dm

```diff
@@ -124,21 +126,51 @@
 	else
 		return ..()
 
+/// Worker proc that checks logic for if a pump can happen, and applies effects/notifications from doing so
+/obj/item/organ/internal/heart/cursed/proc/on_pump(mob/owner)
+	var/next_pump = last_pump + pump_delay - (1 SECONDS) // pump a second early
+	if(world.time < next_pump)
+		to_chat(owner, span_userdanger("Too soon!"))
+		return
+
+	last_pump = world.time
+	playsound(owner,'sound/effects/singlebeat.ogg', 40, TRUE)
+	to_chat(owner, span_notice("Your heart beats."))
+
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/accursed = owner
+
+	if(HAS_TRAIT(accursed, TRAIT_NOBLOOD) || !accursed.dna)
+		return
```

Comment:
Can't we move the HAS_TRAIT() check above the typecast, since traits should work at the datum level? This is just so we can get out of the proc a bit faster.